### PR TITLE
Updated Adcolony Adapter with static GDPR API.

### PR DESCRIPTION
**Pull request by spilaka**

### CODE REVIEW

File: ThirdPartyAdapters/adcolony/adcolony/src/main/java/com/google/ads/mediation/adcolony/AdColonyMediationAdapter.java [hunk 1]
```diff
@@ -30,6 +30,8 @@
 public class AdColonyMediationAdapter extends Adapter implements MediationRewardedAd {
     private static final String TAG = AdColonyMediationAdapter.class.getSimpleName();
 
+    public static AdColonyAppOptions appOptions;
```
Comment (ericleich) [inline ThirdPartyAdapters/adcolony/adcolony/src/main/java/com/google/ads/mediation/adcolony/AdColonyMediationAdapter.java]: Make this variable private and use a getter

File: ThirdPartyAdapters/adcolony/adcolony/src/main/java/com/google/ads/mediation/adcolony/AdColonyMediationAdapter.java [hunk 2]
```diff
@@ -185,6 +189,12 @@ public void loadRewardedAd(
         }
     }
 
+    public static void setAppOptions(AdColonyAppOptions options) {
+        appOptions = new AdColonyAppOptions();
```
Comment (ericleich) [inline ThirdPartyAdapters/adcolony/adcolony/src/main/java/com/google/ads/mediation/adcolony/AdColonyMediationAdapter.java]: don't create a new one. Just set the local variable.

File: ThirdPartyAdapters/adcolony/adcolony/src/main/java/com/jirbo/adcolony/AdColonyManager.java [hunk 3]
```diff
@@ -147,20 +149,19 @@ public boolean configureAdColony(MediationRewardedAdConfiguration adConfiguratio
      */
     private AdColonyAppOptions buildAppOptions(MediationAdRequest adRequest,
                                                Bundle networkExtras) {
-        AdColonyAppOptions options = new AdColonyAppOptions();
+        AdColonyAppOptions options;
```
Comment (ericleich) [inline ThirdPartyAdapters/adcolony/adcolony/src/main/java/com/jirbo/adcolony/AdColonyManager.java]: AdColonyAppOptions options = AdColonyMediationAdapter.getAppOptions();
if (options == null) {
  optiions = new AdColonyAppOptions();
}

File: ThirdPartyAdapters/adcolony/adcolony/src/main/java/com/jirbo/adcolony/AdColonyManager.java [hunk 4]
```diff
@@ -210,20 +211,19 @@ private AdColonyAppOptions buildAppOptions(MediationAdRequest adRequest,
      */
     private AdColonyAppOptions buildAppOptions(MediationRewardedAdConfiguration adConfiguration) {
         Bundle networkExtras = adConfiguration.getMediationExtras();
-        AdColonyAppOptions options = new AdColonyAppOptions();
+        AdColonyAppOptions options;
```
Comment (ericleich) [inline ThirdPartyAdapters/adcolony/adcolony/src/main/java/com/jirbo/adcolony/AdColonyManager.java]: set it to the appOptions by default. If it's still null, set it to a new instance.

**Comment by ericleich** [comment]
@brittmullen FYI

File: ThirdPartyAdapters/adcolony/adcolony/src/main/java/com/google/ads/mediation/adcolony/AdColonyMediationAdapter.java [hunk 5]
```diff
@@ -29,6 +29,7 @@
 
 public class AdColonyMediationAdapter extends Adapter implements MediationRewardedAd {
     private static final String TAG = AdColonyMediationAdapter.class.getSimpleName();
+    private static AdColonyAppOptions appOptions;
```
Comment (ericleich) [inline ThirdPartyAdapters/adcolony/adcolony/src/main/java/com/google/ads/mediation/adcolony/AdColonyMediationAdapter.java]: Can we initialize this to new AppOptions(), and remove the setter?

File: ThirdPartyAdapters/adcolony/adcolony/src/main/java/com/google/ads/mediation/adcolony/AdColonyMediationAdapter.java [hunk 6]
```diff
@@ -112,7 +113,9 @@ public void initialize(Context context,
         }
 
         // Always set mediation network info.
-        AdColonyAppOptions appOptions = new AdColonyAppOptions();
+        if (appOptions == null) {
+            appOptions = new AdColonyAppOptions();
```
Comment (ericleich) [inline ThirdPartyAdapters/adcolony/adcolony/src/main/java/com/google/ads/mediation/adcolony/AdColonyMediationAdapter.java]: No need for this if we initialize at the top, and remove the setter so nobody can set it to null.

File: ThirdPartyAdapters/adcolony/adcolony/src/main/java/com/google/ads/mediation/adcolony/AdColonyMediationAdapter.java [hunk 7]
```diff
@@ -185,6 +188,15 @@ public void loadRewardedAd(
         }
     }
 
+    public static void setAppOptions(AdColonyAppOptions options) {
```
Comment (ericleich) [inline ThirdPartyAdapters/adcolony/adcolony/src/main/java/com/google/ads/mediation/adcolony/AdColonyMediationAdapter.java]: Remove setter.

File: ThirdPartyAdapters/adcolony/adcolony/src/main/java/com/jirbo/adcolony/AdColonyBundleBuilder.java [hunk 8]
```diff
@@ -11,8 +11,6 @@
     private static boolean _showPreAdPopup;
     private static boolean _showPostAdPopup;
     private static String _userId;
```
Comment (ericleich) [inline ThirdPartyAdapters/adcolony/adcolony/src/main/java/com/jirbo/adcolony/AdColonyBundleBuilder.java]: Looking at this, we should get rid of the entire AdColonyBundleBuilder class, if all of these params can be set directly on the AppOptions class that publishers now have access to.

Comment (spilaka) [reply ThirdPartyAdapters/adcolony/adcolony/src/main/java/com/jirbo/adcolony/AdColonyBundleBuilder.java]: Not removed bundle builder. As we discussed, to pass only gdprRequired and gdprConsentString before init.

**Comment by ericleich** [comment]
Reason for CLA: Yes. Commit author is covered via separate terms. Also has @google.com email.

**Pull request merged**